pkg/management/component.go

```diff
@@ -661,6 +663,22 @@ func (c *component) ResetSmsCounter(ctx context.Context, realmName, userID strin
 	return nil
 }
 
+func (c *component) RecoveryCode(ctx context.Context, realmName, userID string) (string, error) {
+	var accessToken = ctx.Value(cs.CtContextAccessToken).(string)
+
+	recoveryCodeKc, err := c.keycloakClient.CreateRecoveryCode(accessToken, realmName, userID)
+
+	if err != nil {
+		c.logger.Warn(ctx, "err", err.Error())
+		return "", err
+	}
+
+	// store the API call into the DB
+	c.reportEvent(ctx, "RECOVERY_CODE", database.CtEventRealmName, realmName, database.CtEventUserID, userID)
```

Comment:
CREATE_RECOVERY_CODE ?

Reply:
Done.
Also renamed the methods with CreateRecoveryCode everywhere